### PR TITLE
[DO NOT MERGE] Re-factor code corresponding to #41 issue in Nailgun, part 2

### PR DESCRIPTION
**Pull request by abalakh**
Current fix should be merged right after the SatelliteQE/nailgun#166 gets to the master branch
Tests results:
```
nosetests tests/foreman/api/test_contentview.py -m test_positive_publish
..
----------------------------------------------------------------------
Ran 2 tests in 201.148s

OK

nosetests tests/foreman/api/test_product.py -m test_repositoryset_
..
----------------------------------------------------------------------
Ran 2 tests in 135.433s

OK

nosetests tests/foreman/api/test_repository.py -m test_update_contents
.
----------------------------------------------------------------------
Ran 1 test in 20.180s

OK

nosetests tests/foreman/api/test_repository.py -m test_redhat_sync_1 
.
----------------------------------------------------------------------
Ran 1 test in 86.361s

OK

nosetests tests/foreman/api/test_syncplan.py -m test_add           
..
----------------------------------------------------------------------
Ran 2 tests in 28.309s

OK

nosetests tests/foreman/api/test_syncplan.py -m test_remove
S.
----------------------------------------------------------------------
Ran 2 tests in 24.830s

OK (SKIP=1)

nosetests tests/foreman/api/test_syncplan.py -m test_delete              
...
----------------------------------------------------------------------
Ran 3 tests in 49.883s

OK

nosetests tests/foreman/ui/test_activationkey.py -m test_associate_product_3
.
----------------------------------------------------------------------
Ran 1 test in 144.799s

OK

nosetests tests/foreman/ui/test_contentviews.py -m test_associate_puppet_module
.
----------------------------------------------------------------------
Ran 1 test in 83.004s

OK

nosetests tests/foreman/smoke/test_api_smoke.py -m test_end_to_end            
.
----------------------------------------------------------------------
Ran 1 test in 390.575s

OK
```

### CODE REVIEW

**Comment by abalakh** [comment]
Re-ran all the api tests:
```
nosetests tests/foreman/api                                      
......SSS.....................S.............................................SSSSSSSSSSSSSSSSSSSSSSS...........................................................................................SSSSSSSSSSSSSSSSSSSSSSSSSEEEEEE......E.SSSSSSSSSSSSS.............S.SSSSSSS.SSSSSS................................................S.........SSSSSSSSSSSSS...S..S...S..SSSSSSSSSS...................................................SSSSSSSS................S.................................................................................................................................................................S.............................................................SSSSSS..S.SS.SSSSS.S..SSS.SSSSSSSSSSSSSSSSSSSSSSSSSSSSS.S..................S..S...............SSSSSS..............................................................................................................S..........................................................SSSSSSSS.....................................................S.SSS.........................................................................................
----------------------------------------------------------------------
Ran 1088 tests in 9579.628s

FAILED (SKIP=185, errors=7)
```
(7 failures are docker tests, i don't have docker urls configured in my robottelo.properties)

File: tests/foreman/api/test_product.py [hunk 1]
```diff
@@ -171,25 +167,17 @@ def test_repositoryset_disable(self):
...
-        repositories = product.repository_sets_available_repositories(
-            reposet_id=reposet_id,
-        )
+            organization=org,
+        ).search()[0]
+        reposet = entities.RepositorySet(
+            name=REPOSET['rhva6'],
+            product=product,
+        ).search()[0]
+        reposet.enable({'basearch': 'x86_64', 'releasever': '6Server'})
```
Comment (sthirugn) [inline tests/foreman/api/test_product.py]: As per meeting discussion today, Can we create a robottelo helper method for this?

File: tests/foreman/api/test_product.py [hunk 2]
```diff
@@ -171,25 +167,17 @@ def test_repositoryset_disable(self):
...
+        reposet.disable({'basearch': 'x86_64', 'releasever': '6Server'})
```
Comment (sthirugn) [inline tests/foreman/api/test_product.py]: As per meeting discussion today, Can we create a robottelo helper method for this?

**Comment by sthirugn** [comment]
ACK pending comments.

**Comment by sthirugn** [comment]
Creating helper methods is non blocker, we can do this later as well.

File: tests/foreman/api/test_product.py (hunk 1 above)
Comment (abalakh) [inline tests/foreman/api/test_product.py]: Should that helper func work with reposet/product ids? Or it should search by name instead?
I'm not sure what's more common use case. And AFAIK we don't usually have product/reposet ids and often performing searching for that, right?

Comment (sthirugn) [inline tests/foreman/api/test_product.py]: I am not sure at this point either.  We may need to scan through robottelo test cases to find this.  As I commented earlier, not a blocker for this PR, but definitely worth looking into.

**Comment by oshtaier** [comment]
Merge only in case nailgun PRs merged into master!

File: tests/foreman/api/test_product.py (hunk 1 above)
Comment (abalakh) [inline tests/foreman/api/test_product.py]: IMO, if we have cases when we have only reposet & product name and cases when we have their IDs, helper func should be more factory-like and should work in both situations. But yeah, that's different issue, the purpose of this PR is to make robottelo work with beta version of nailgun. So I've submitted issue #2638 for creating helper functions.

Comment (sthirugn) [inline tests/foreman/api/test_product.py]: :+1:

**Comment by Ichimonji10** [comment]
See SatelliteQE/nailgun#171.

**Comment by Ichimonji10** [comment]
I'll pick up this PR and see what I can do to make it compatible with SatelliteQE/nailgun#171.

**Comment by oshtaier** [comment]
There should not be so many changes as most of the issues should be in 1 part

**Comment by Ichimonji10** [comment]
@oshtaier Does this PR depend on part 1?

**Comment by oshtaier** [comment]
@Ichimonji10 nope, it should not

**Comment by Ichimonji10** [comment]
Superceded by #2658.

**Comment by Ichimonji10** [comment]
Included in #2658.

**Pull request closed without merging**